src/core/task/index.ts

```diff
@@ -306,10 +311,10 @@ export class Task {
 		// initialize telemetry
 		if (historyItem) {
 			// Open task from history
-			telemetryService.captureTaskRestarted(this.taskId, currentProvider)
+			telemetryService.captureTaskRestarted(this.taskId, this.uuid, currentProvider)
```

Comment:
Rather than a custom UUID, it would be better to follow a published standard like W3C Trace Context (https://www.w3.org/TR/trace-context/).

This would allowed distributed traces, i.e. passing the trace identifier in outgoing requests (such as web requests), allowing cross-tier calls to be trace if necessary. The W3C Trace ID format is a standard industry format that, along with OpenTelemetry, has been adopted by every major telemetry provider that I can think of.

Reply:
Thanks for the mention and we could definitely consider it down the line.